### PR TITLE
Add DevWidgets

### PR DESCRIPTION
**Pull request by gumbarros**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: br.com.barros.DevWidgets.desktop [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=DevWidgets
+Version=1.0
+Comment=DevWidgets is a Flutter application with several tools such as generators, formatters and converters for developers. It's directly inspired by DevToys. 
+Exec=devwidgets %u
+Icon=icon
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.desktop:6]: icon name is invalid.

File: br.com.barros.DevWidgets.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{
+    "app-id": "br.com.barros.DevWidgets",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "21.08",
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.json:4]:
```suggestion
    "runtime-version": "22.08",
```

File: br.com.barros.DevWidgets.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      "--device=dri",
+      "--socket=pulseaudio",
+      "--share=network"
+    ],
+    "modules": [
+      {
+        "name": "devwidgets",
+        "buildsystem": "simple",
+        "only-arches": [
+          "x86_64"
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.json:21]: this doesn't do what you think it does. You are still trying to build an aarch64 flatpak from this.

File: br.com.barros.DevWidgets.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          "mkdir -p /app/bin",
+          "ln -s /app/DevWidgets/devwidgets /app/bin/devwidgets",
+          "mkdir -p /app/share/metainfo",
+          "cp -r br.com.barros.DevWidgets.metainfo.xml /app/share/metainfo/",
+          "mkdir -p /app/share/icons/hicolor/scalable/apps",
+          "cp -r icon.svg /app/share/icons/hicolor/scalable/apps/br.com.barros.DevWidgets.svg",
+          "mkdir -p /app/share/applications",
+          "cp -r br.com.barros.DevWidgets.desktop /app/share/applications/",
+          "mkdir -p /app/share/appdata",
+          "cp -r br.com.barros.DevWidgets.metainfo.xml /app/share/appdata/"
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.json:37]:
```suggestion
          "install -Dm644 br.com.barros.DevWidgets.metainfo.xml  -t /app/share/metainfo/",
          "install -Dm644 icon.svg /app/share/icons/hicolor/scalable/apps/br.com.barros.DevWidgets.svg",
          "install -Dm644 br.com.barros.DevWidgets.desktop -t /app/share/applications/"
```

File: br.com.barros.DevWidgets.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            "type": "file",
+            "path": "br.com.barros.DevWidgets.desktop"
+          },
+          {
+            "type": "file",
+            "path": "icon.svg"
+          },
+          {
+            "type": "file",
+            "path": "DevWidgets-Linux-Portable.tar.gz"
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.json:54]: definitely not. 
1. this file is not in the repo.
2. this file **SHOULD be downloaded from somewhere** and not be in the repo anyway.

File: br.com.barros.DevWidgets.metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  <project_license>MIT</project_license>
+  <supports>
+    <control>pointing</control>
+    <control>keyboard</control>
+    <control>touch</control>
+  </supports>
+  <description>
+    <p>DevWidgets is a Flutter application with several tools such as generators, formatters and converters for developers. It's directly inspired by DevToys. </p>
+  </description>
+  <launchable type="desktop-id">br.com.DevWidgets.desktop</launchable>
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.metainfo.xml:23]: this doesn't match the desktop file name

File: br.com.barros.DevWidgets.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- 
+    Generator for metainfo & .desktop files: 
+    https://www.freedesktop.org/software/appstream/metainfocreator/#/
+-->
+<component type="desktop-application">
+  <id>br.com.barros.DevWidgets</id>
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.metainfo.xml:8]: I am pretty sure that `barros.com.br` is UNRELATED to this. So the app-id doesn't meet the requirements.

**Review by hfiguiere — CHANGES_REQUESTED** [review]
.desktop. metainfo and icon should be submitted and managed upstream (for at least a future release)

As is it stands now this won't build.

File: br.com.barros.DevWidgets.json [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            "type": "file",
+            "path": "br.com.barros.DevWidgets.metainfo.xml"
+          },
+          {
+            "type": "file",
+            "path": "br.com.barros.DevWidgets.desktop"
+          },
+          {
+            "type": "file",
+            "path": "icon.svg"
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.json:50]: the icon is not square. This will fail.

File: br.com.barros.DevWidgets.json [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    "modules": [
+      {
+        "name": "devwidgets",
+        "buildsystem": "simple",
+        "only-arches": [
+          "x86_64"
+        ],
+        "build-commands": [
+          "mkdir -p DevWidgets",
+          "tar -xf DevWidgets-Linux-Portable.tar.gz -C DevWidgets",
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.json:25]: this should be made as source of type `archive` and let flatpak-builder extract it.

File: br.com.barros.DevWidgets.desktop (hunk 1 above)
Comment (gumbarros) [reply br.com.barros.DevWidgets.desktop:6]: Should be icon.svg?

File: br.com.barros.DevWidgets.json (hunk 5 above)
Comment (gumbarros) [reply br.com.barros.DevWidgets.json:49]: I'm really new to flatpak, how can I make it download from my github repo?

File: br.com.barros.DevWidgets.json (hunk 3 above)
Comment (gumbarros) [reply br.com.barros.DevWidgets.json:21]: What is the correct form? My intent is to build for x64

File: br.com.barros.DevWidgets.desktop [hunk 10]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Name=DevWidgets
+Version=1.0
+Comment=Utilities for developers. 
+Exec=devwidgets %u
+Icon=icon.svg
```
Comment (hfiguiere) [inline br.com.barros.DevWidgets.desktop:6]: this is still incorrect. Name never include the extension.

See how you install the icon for the name.

**Pull request closed without merging**